examples/identity_hasher.rs

```diff
@@ -0,0 +1,114 @@
+use bytes::{Buf, Bytes};
+
+use multihash::derive::Multihash;
+use multihash::{Error, Hasher, MultihashDigest, MultihashGeneric, Sha2_256};
+
+/// update appends and truncates/ignores updates > S
+#[derive(Debug)]
+pub struct IdentityTrunk<const S: usize> {
+    cursor: usize,
+    arr: [u8; S],
+}
+
+impl<const S: usize> Default for IdentityTrunk<S> {
+    fn default() -> Self {
+        Self {
+            cursor: 0,
+            arr: [0u8; S],
+        }
+    }
+}
+impl<const S: usize> Hasher for IdentityTrunk<S> {
+    fn update(&mut self, input: &[u8]) {
+        let src_end = (self.cursor + input.len()).min(self.arr.len());
+        let input_end = input.len().min(self.arr.len() - self.cursor);
+
+        self.arr[self.cursor..src_end].copy_from_slice(&input[..input_end]);
+        self.cursor = src_end;
+    }
+    fn finalize(&mut self) -> &[u8] {
+        &self.arr[..self.cursor]
+    }
+    fn reset(&mut self) {
+        *self = Self {
+            cursor: 0,
+            arr: [0u8; S],
+        };
+    }
+}
+
+/// Borrows and concats slices using Bytes crate
+#[derive(Debug, Default)]
+pub struct IdentityBorrow(Bytes);
+
+impl Hasher for IdentityBorrow {
+    fn update(&mut self, input: &[u8]) {
+        let new = (*self.0).chain(input);
+        self.0 = Bytes::copy_from_slice(new.chunk());
+    }
+    fn finalize(&mut self) -> &[u8] {
+        &self.0
+    }
+    fn reset(&mut self) {
+        *self = Self::default();
+    }
+}
+
+/// update allocates & appends, Alloc only
+#[derive(Default, Debug)]
+pub struct IdentityAlloc(Vec<u8>);
+
+impl Hasher for IdentityAlloc {
+    fn update(&mut self, input: &[u8]) {
+        self.0.extend(input);
+    }
+    fn finalize(&mut self) -> &[u8] {
+        &self.0
+    }
+    fn reset(&mut self) {
+        self.0.clear();
+    }
+}
+
+#[derive(Clone, Copy, Debug, Eq, Multihash, PartialEq)]
+#[mh(alloc_size = 64)]
+pub enum Code {
+    #[mh(code = 0x00, hasher = IdentityTrunk::<64>)]
+    IdentityTrunk,
+    // These will panic and if input > alloc_size, use alternate methods if you want to use bigger than alocated size
+    // #[mh(code = 0x02, hasher = IdentityBorrow)]
+    // IdentityBorrow,
+    // #[mh(code = 0x03, hasher = IdentityAlloc)]
+    // IdentityAlloc,
```

Comment:
In case you also want to add examples for those, you could use a code from the private area of the multicodec table (as you onviously can't use `0x00` for all of them: https://github.com/multiformats/multicodec/tree/f5dd49f35b26b447aa380e9a491f195fd49d912c#private-use-area

Reply:
Borrow and Alloc examples were for me to play around with the concept of integrating. The alloc one specifically is funny since while it will hash the entirety of the input, it still gets shoved into the default sized array and crashes. this could be fixed by leaning into traits more and splitting `MultishashStack<S>` and `MultihashAlloc` into their own structs. However, good alloc support is not a good focus for this crate and would only apply to Identity hashing and maybe Blake3 with its extendable output.

I will the remove comments and references to them in the example, though if you want I can open a tracking issue about better alloc support to hopefully focus the attention of people that want it.